### ISSUE TITLE
Style/RedundantLineContinuation correction creates invalid syntax

### ISSUE THREAD

**Issue by jdufresne** [rubocop/rubocop#11778]
Example input:

```rb
def f
  {
    k: @v.p1? \
      && !@v.p2? \
      && !@v.p3?
  }
end
```

## Expected behavior

Autocorrect should fix the Style/RedundantLineContinuation offense without creating invalid syntax.

Potential correction:

```rb
def f
  {
    k: @v.p1? && !@v.p2? && !@v.p3?
  }
end
```

Or

```rb
def f
  {
    k: @v.p1? &&
      !@v.p2? &&
      !@v.p3?
  }
end
```

## Actual behavior

```
$ bundle exec rubocop --debug -A test.rb
For /home/jon/test: configuration from /home/jon/test/.rubocop.yml
Default configuration from /home/jon/test/vendor/bundle/ruby/3.2.0/gems/rubocop-1.49.0/config/default.yml
Use parallel by default.
Skipping parallel inspection: only a single file needs inspection
Inspecting 1 file
Scanning /home/jon/test/test.rb
Loading cache from /home/jon/.cache/rubocop_cache/1112ce6db0fa8401f299b5df780370478b4b27c0/6d7a3b621ca1730e04accd938619e4bdab66cfb1/b6bd7f4a413c48a818f53cbcb29c19e71032b957
F

Offenses:

test.rb:6:18: C: [Corrected] Style/RedundantLineContinuation: Redundant line continuation.
      && !@v.p2? \ ...
                 ^
test.rb:7:7: F: Lint/Syntax: unexpected token tAMPER
(Using Ruby 2.6 parser; configure using TargetRubyVersion parameter, under AllCops)
      && !@v.p3?
      ^
test.rb:8:3: F: Lint/Syntax: unexpected token tRCURLY
(Using Ruby 2.6 parser; configure using TargetRubyVersion parameter, under AllCops)
  }
  ^

1 file inspected, 3 offenses detected, 1 offense corrected
Finished in 0.08526759898813907 seconds
```

## Steps to reproduce the problem

Use the example input file provided above with the configuration:

```yml
AllCops:
  NewCops: enable

Metrics:
  Enabled: false

Style/Documentation:
  Enabled: false

Style/FrozenStringLiteralComment:
  Enabled: false
```

## RuboCop version

```
$ bundle exec rubocop -V
1.49.0 (using Parser 3.2.2.0, rubocop-ast 1.28.0, running on ruby 3.2.0) [x86_64-linux]
```

**Comment by koic** [comment]
This issue is a dup with #11756.
